src/core/ucc_constructor.c

```diff
@@ -53,6 +55,19 @@ ucc_status_t ucc_constructor(void)
             get_default_lib_path();
         }
     }
+
+    if (!ucc_global_config.component_path) {
+        //TODO
+        //ucc_error, "Failed to get ucc library path. set UCC_COMPONENT_PATH.\n");
+        return UCC_ERR_NO_MESSAGE;
+    }
+    if (UCC_ERR_NO_MESSAGE ==
+        ucc_components_load("cl", &ucc_global_config.cl_framework)) {
```

Comment:
shouldn't "cl" be some kind of parameter? I assume we'll need the same for "tl", "topo" and others...

Reply:
HEre in constructor we are actually loading "CL" components, this is why passing "cl" as parameter to the component framework loader. So, it is a parameter. At some point we will add similar "ucc_components_load" line for "tl" as well.